composer/datasets/in_context_learning_evaluation.py

```diff
@@ -933,173 +765,83 @@ class InContextLearningCodeEvalDataset(Dataset):
 
     def __init__(
         self,
-        dataset_uri: str,
-        tokenizer: Union[transformers.PreTrainedTokenizer, transformers.PreTrainedTokenizerFast],
-        max_seq_len: int,
-        pad_tok_id: int,
-        num_fewshot: int,
-        prompt_string: str,
-        example_delimiter: str,
-        destination_path: str,
-        code_prelimiter: str,
-        fewshot_random_seed: int,
         generations_per_sample: int,
         pass_at_k: int = 1,
         top_p: Optional[float] = 0.95,
         top_k: Optional[int] = 40,
```

Comment:
should we add temperature?

Reply:
Yes we should. Can add in this PR but felt like I should focus on the refactor not more features. If you give me a thumbs up I can add it.